tests/test_parsing_priorities.py

```diff
@@ -0,0 +1,94 @@
+import json
+import sys
+
+import valohai
+from valohai.internals.global_state_loader import flush_global_state
+from valohai.internals.inputs import get_input_info
+
+# Inputs and parameters can come from (in the order of priority):
+#
+# 1. Command-line
+# 2. parameters.json and inputs.json configs
+# 3. Call to valohai.prepare()
+#
+# Here we test that the priority order is honored
+
+
+def test_parsing_priorities(tmpdir, monkeypatch):
+    local_file = tmpdir.mkdir("sub").join("hello.txt")
+    local_file.write("tiku ja taku ja joku")
+
+    parameters = {
+        "floaty": 0.0001,
+    }
+    inputs = {
+        "example": "https://valohai-mnist.s3.amazonaws.com/t10k-images-idx3-ubyte.gz",
+    }
+
+    with monkeypatch.context() as m:
```

Comment:
You don't need the separate monkeypatch context here, just do `monkeypatch.setenv` and `monkeypatch.setattr`.